library/std/src/io/mod.rs

```diff
@@ -945,6 +945,41 @@ pub trait Read {
     }
 }
 
+/// Read all bytes from a [reader][Read] into a new [`String`].
+///
+/// This is a convenience function for [`Read::read_to_string`]. Using this
+/// function avoids having to create a variable first and provides more type
+/// safety since you can only get the buffer out if there were no errors. (If you
+/// use [`Read::read_to_string`] you have to remember to check whether the read
+/// succeeded because otherwise your buffer will be empty or only partially full.)
+///
+/// # Errors
+///
+/// This function forces you to handle errors because the output (the `String`)
+/// is wrapped in a [`Result`]. See [`Read::read_to_string`] for the errors
+/// that can occur. If any error occurs, you will get an [`Err`], so you
+/// don't have to worry about your buffer being empty or partially full.
+///
```

Comment:
Should I add a section explaining that this is less performant? We can also always add that later if we don't add it now.

Reply:
Yeah, might be good. But I don't think it's very important. You could add something like `However, this does mean the newly allocated buffer will be deallocated again when an error occurs. To re-use the same buffer, you can use Read::read_to_string directly.`, or something in that direction.

Reply:
Well I think the biggest perf issue is that it doesn't let you pre-allocate space (i.e. no `String::with_capacity`). But I can mention both maybe.

Reply:
Added performance docs.